documentation/projects/proposals/publish_dataset/20230706-project_proposal.md

```diff
@@ -0,0 +1,207 @@
+of those offerings we use models hosted on the HuggingFace hub. The
+[Datasets library](https://huggingface.co/docs/datasets/index) allows for easy
+loading of the Openverse dataset in any data pipelines we write. HuggingFace
+also offers the ability to deploy production-ready API endpoints for
+transformation models hosted on their hub. This feature is called
+[Inference Endpoints](https://huggingface.co/docs/inference-endpoints/index).
+
+#### Potentially-related projects
+
+- [External API for sensitive content detection #422](https://github.com/WordPress/openverse/issues/422) -
+  Instead of Rekognition or Google Vision, we may want to use a community model
+  on HuggingFace.
+- [Machine Image Labeling pipeline #426](https://github.com/WordPress/openverse/issues/426)
+  Either a community member may create an up to date set of machine-generated
+  labels we could use, or we could again create an Inference Endpoint with an
+  existing model, for example
+  [Vision Transformer](https://huggingface.co/google/vit-base-patch16-224)
+- [Duplicate image detection #427](https://github.com/WordPress/openverse/issues/427) -
+  Deduplication is an extremely common preprocessing step. We should be able to
+  apply the deduplication from
+
+### Resiliency
+
+The metadata for openly-licensed media, used by Openverse to power the API and
+frontend, is a community utility and should be available to all users,
+_distinctly_ from Openverse itself. By publishing this dataset, we ensure access
+to this data is fast, accessible, and resilient. With published datasets, this
+access remains even if Openverse is inaccessible, under attack, or experiencing
+any other unforeseen disruptions.
+
+## Goals
+
+<!-- Which yearly goal does this project advance? -->
+
+This project encompasses all of our 2023 "lighthouse goals", but "Community
+Development" is perhaps the broadest and most relevant. Others touched on here,
+or impacted through potential downstream changes, are "Result Relevancy",
+"Quantifying our Work","Search Experience", "Content Safety", and "Data
+inertia".
+
+## Requirements
+
+<!-- Detailed descriptions of the features required for the project. Include user stories if you feel they'd be helpful, but focus on describing a specification for how the feature would work with an eye towards edge cases. -->
+
+This project requires coordination with HuggingFace to release the dataset on
+their platform, bypassing their typical restrictions for Dataset size.
+
+We will also need to figure out the technical requirements for producing the raw
+dataset, which will be done in this project's implementation plan. Additionally
+in that plan we will:
+
+- Determine if we host a single dataset for all media types, or separate
+  datasets for different media types.
+- Develop a plan for updating the datasets regularly
+- Refine how we will provide access to our raw dataset for upload to
+  HuggingFace:
+  - Delivery mechanism (a requester pays S3 bucket?)
```

Comment:
On HF side we don't plan on having bottlenecks or paid egress for the dataset. It would be fully free to access and download the entire dataset or parts of it via streaming. 

If the final dump size in the final format doesn't fit our traditional datasets repos - we can provide infra for it to be always openly accessible